src/vs/platform/native/electron-main/nativeHostMainService.ts

```diff
@@ -812,6 +812,27 @@ export class NativeHostMainService extends Disposable implements INativeHostMain
 
 	async setPassword(windowId: number | undefined, service: string, account: string, password: string): Promise<void> {
 		const keytar = await this.withKeytar();
+		const MAX_SET_ATTEMPTS = 5;
```

Comment:
IMO 6 attempts is a bit much, I think if it doesn't succeed after 3 tries, it's probably just not going to succeed. So I suggest making this 2

Reply:
Especially since that introduces a perceptible wait time

Reply:
Yeah I don't might changing it to 3. I can even shorten the MS to like 200 so it all still operates under 1s.

Reply:
set to 3.